p2p/src/main/java/bisq/network/p2p/network/Connection.java

```diff
@@ -280,31 +280,16 @@ public void sendMessage(NetworkEnvelope networkEnvelope) {
     }
 
     public boolean noCapabilityRequiredOrCapabilityIsSupported(Proto msg) {
-        return !isCapabilityRequired(msg) || isCapabilitySupported(msg);
-    }
-
-    @SuppressWarnings("BooleanMethodIsAlwaysInverted")
-    public static boolean isCapabilityRequired(Proto msg) {
-        if (msg instanceof AddDataMessage) {
-            final ProtectedStoragePayload protectedStoragePayload = (((AddDataMessage) msg).getProtectedStorageEntry()).getProtectedStoragePayload();
-            return protectedStoragePayload instanceof CapabilityRequiringPayload;
-        } else if (msg instanceof AddPersistableNetworkPayloadMessage) {
-            final PersistableNetworkPayload persistableNetworkPayload = ((AddPersistableNetworkPayloadMessage) msg).getPersistableNetworkPayload();
-            return persistableNetworkPayload instanceof CapabilityRequiringPayload;
-        } else {
-            return msg instanceof CapabilityRequiringPayload;
-        }
-    }
-
-    private boolean isCapabilitySupported(Proto msg) {
         if (msg instanceof AddDataMessage) {
             final ProtectedStoragePayload protectedStoragePayload = (((AddDataMessage) msg).getProtectedStorageEntry()).getProtectedStoragePayload();
             return protectedStoragePayload instanceof CapabilityRequiringPayload && sharedModel.isCapabilitySupported(((CapabilityRequiringPayload) protectedStoragePayload).getRequiredCapabilities());
         } else if (msg instanceof AddPersistableNetworkPayloadMessage) {
             final PersistableNetworkPayload persistableNetworkPayload = ((AddPersistableNetworkPayloadMessage) msg).getPersistableNetworkPayload();
             return persistableNetworkPayload instanceof CapabilityRequiringPayload && sharedModel.isCapabilitySupported(((CapabilityRequiringPayload) persistableNetworkPayload).getRequiredCapabilities());
+        } else if(msg instanceof CapabilityRequiringPayload) {
+            return sharedModel.isCapabilitySupported(((CapabilityRequiringPayload) msg).getRequiredCapabilities());
         } else {
-            return msg instanceof CapabilityRequiringPayload && sharedModel.isCapabilitySupported(((CapabilityRequiringPayload) msg).getRequiredCapabilities());
+            return true;
```

Comment:
I thought about changing stuff around but it quickly gets dirty.

Most beautiful would be that we do not need the method at all but instead can use a `connection.isCapabilitySupported(msg)` or `isCapabilitySupported(msg)` directly. However, the class hierarchy has grown overly complicated over time and such a solution is not possible as things stand now (would result in a diamond pattern if Java would allow for that).

Even looks worse if depicted like that:
![messages](https://user-images.githubusercontent.com/1070734/53952939-6eef5800-40d2-11e9-8ca4-1059082d07d9.png)

I even tried to use java8/9 defaults in interfaces and create some sort of mixins. Nasty stuff. Definitely needs cleanup! Small steps...